### DISCUSSION TITLE
AR on Android is crashing

### DISCUSSION THREAD

**Discussion by KarimSalam9** [google/model-viewer#4681] [Q&A] [answered]
Hello, 

I currently run my projects in babylonjs, i used to use the model viewer ar by sending the scene into a blob and passing it to the model viewer source, initially it was normally functioning, suddenly it stopped working on android only on ios its still working normally, so when i press to got to AR on android devices it switches to the ar app then suddenly crashes before even loading the model, whereas if i load the model directly from the public folder locally to the model viewer source it works and doesn't crash, i need a solution for it to work on android through sending it in a blob url since it has to be dynamic with customizations and whatever else, any solution please ?

**Comment by elalish** [comment]
It sounds like the problem is that WebXR has stopped working for some reason. Our Scene Viewer AR mode was never able to work with blobs (it's a separate app outside our sandbox). Check your JS console and you may find some warnings about why WebXR failed to start. It may be related to this: https://github.com/google/model-viewer/issues/4582 which appears to be a WebXR regression, but it's been difficult to repro or debug. Any more detailed info you can give would be helpful.

**Comment by KarimSalam9** [comment]
Thanks for your reply, the weird thing i checked the console i connected the phone and used the debugger, it says that its starting the AR and then no errors or warnings appear, i tried using the ar mode as webxr, when i did that it didnt even initialize the ar i just click the button and nothing happens

`<model-viewer
      style={{ width: "90%", height: "80%" }}
      //   ar-modes="quick-look"
      // ar-tracking="tracking"
      src={glbmodel}
      // alt="Description of your 3D model"
      //   auto-rotate  n
      //   camera-controls
      // ar-placement="floor"
      //   ar-scale="fixed"
      // touch-action="pan-x"
      ar
    ></model-viewer>`
    
    this is the code for the model viewer the glbmodel im sending it as a state from the scene3d, so when i press the button i export the model and send a blob url of it to the modelviewer, the thing is it used to work but a while ago it stopped working any clue ?

**Reply by elalish** [reply]
`ar-modes="quick-look"` only enables AR on iOS - if you want Android, you'll need `ar-modes="webxr quick-look"`

**Reply by KarimSalam9** [reply]
Yes here it is commented, what i tried is setting ar-modes="webxr" and nothing happened, does it work like that or should i include quick-look with it ?

**Reply by elalish** [reply]
Yeah, it should work like that too - anything in the JS console? You're on HTTPS, right?

**Reply by KarimSalam9** [reply]
Yes im hosting on netlify, the weird thing is that it requires permission and then does nothing and stays in the scene

**Comment by KarimSalam9** [comment]
Here is an example of whats happening and a screenshot of the console, and the link is : (url)](https://ar-testing-modular.netlify.app/)

![image (5)](https://github.com/google/model-viewer/assets/98595184/ddb32551-c271-4037-b680-5a20d7f9815f)

https://github.com/google/model-viewer/assets/98595184/5cfffa6d-b4e0-46a7-bcef-ef3ff831179e

**Reply by elalish** [reply]
What device, android and browser versions are you using?

**Reply by KarimSalam9** [reply]
I tried on several devices this one is: 

- infinix note 12 vip
- infinix x672
- android version 12
- chrome application version 121.0.6167.164

i also tried it on :

- s21+ 5G one ui version 6.0
- android version 14
- chrome app version Chrome 121.0.6167.178

**Reply by elalish** [reply]
@bialpio some more devices reproducing this WebXR failure to start. Have you managed to get a local repro yet?

**Reply by bialpio** [reply, marked as answer]
I'm unable to reproduce on https://modelviewer.dev/examples/augmentedreality/index.html using Pixel 8, S8+, and S20+5G. I'm using Chrome 121.0.6167.178. When trying on S8+ with https://ar-testing-modular.netlify.app/, I see similar behavior as described in #4665 ("useProgram: program not valid" warning in the console). When testing the netlify page on Pixel 8, I initially had the same (or very similar) results as the OP - the device would enter fullscreen but AR session contents were not visible. The content of the session started to show up only when I changed the element style of `<div class="modelviewerdiv">` to "display: hidden" in dev tools (actually, they started to be visible when I just typed in "hi", as in "display: hi" - I think it was initially "display: none").

Given that things work fine on modelviewer.dev on 2 out of 3 devices (and S8+ seems to render the shadows of the models in AR, suggesting that AR session is fine), it looks like it may be some problem with the netlify.app page.

**Reply by KarimSalam9** [reply]
So i tried the combination of using webxr and removing the display none on the div and it worked ! Thanks for your support

**Comment by crossterra-agent** [comment]
- **Status**: DONE
- **Role**: orchestrator
- **Step**:
- **Link**: https://github.com/CrossTerra/monorepo/pull/1818#issuecomment-3979296249
- **Summary**: Developer posted new DONE (all steps complete, CI passing at 06:42Z); re-invoked reviewer on PR #1818 to verify final state.
